b2share/modules/records/errors.py

```diff
@@ -62,12 +63,28 @@ class AnonymousDepositSearch(B2ShareRecordsError):
 def register_error_handlers(app):
     @app.errorhandler(ValidationError)
     def handle_validation_error(err):
-        field = '/'.join([str(x) for x in err.path])
+        fieldpath = '/'.join([str(x) for x in err.path])
+        message = err.message
+        field = None
         if err.validator == 'required' or err.validator == 'additionalProperties':
             try:
                 field = err.message.split('\'')[1]
+                fieldpath = (fieldpath + '/' if fieldpath else '') + field
             except IndexError:
                 pass # ignore
+        if err.validator == 'required' and len(err.path) == 1 and \
+                err.path[0] == 'community_specific' and is_valid_uuid(field):
+            message = 'The "community_specific" metadata object must contain '\
+                      'an object named "{}" containing the '\
+                      'community-specific metadata fields'.format(field)
```

Comment:
It would be interesting to have a test checking that the message has been changed properly. Maybe I missed it.